### PR TITLE
jre_minimal: convert `passthru.tests` to an attribute set and fix pre/post hook running

### PR DESCRIPTION
**Pull request by ShamrockLee**
## Description of changes

Convert jre_minimal.tests from  a list to an attribute set, making it easier to invoke each one individually.

Fix `runHook preBuildPhase` -> `runHook preBuild` and sort like that for package tests.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ShamrockLee** [comment]
I rebased the changes on top of the `master` branch and resolved the merge conflict by moving the release note entry to `rl-2411.section.md`.

**Comment by ShamrockLee** [comment]
Rebased and resolved the merge conflicts.

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -54,6 +54,8 @@
 
 - `buildbot` was updated to 4.0, the AngularJS frontend has been replaced by a React frontend, see the [upstream release notes](https://docs.buildbot.net/current/manual/upgrading/4.0-upgrade.html).
 
+- `jre_minimal.tests` is now an attribute set of package tests instead of a list, making it easier to access individual test cases. This change does not affect the CI, since `nix-build jre_minimal.tests` still works.
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2411.section.md:57]: I don't think this is noteworthy at all.

Comment (ShamrockLee) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: I had thought it would be considered breaking.

Comment (ShamrockLee) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: Addressed.

**Pull request merged**